kernel/Kconfig.sched_server

```diff
@@ -0,0 +1,130 @@
+# config options for Scheduling Servers
+#
+# Copyright (c) 2024 Instituto Superior de Engenharia do Porto (ISEP)
+# SPDX-License-Identifier: Apache-2.0
+
+menu "Constant Bandwidth Server (CBS)"
+
+config CBS
+	bool "Constant Bandwidth Server (CBS)"
+	depends on USE_SWITCH
+	select SCHED_DEADLINE
+	default n
+	help
+		Enables the Constant Bandwidth Server (CBS), an extension
+		of the Earliest Deadline First (EDF) scheduler that allows
+		tasks to execute with guaranteed bandwidth. Selecting
+		this option automatically enables the EDF scheduler
+		(CONFIG_SCHED_DEADLINE=y). The CBS is also currently
+		supported only in targets that use the _arch_switch
+		context switching primitive (CONFIG_USE_SWITCH=y).
+
+config CBS_LOG
+	bool "CBS event logging"
+	depends on CBS
+	select LOG
+	default n
+	help
+		Enables logging of Constant Bandwidth Server (CBS) events,
+		such as jobs being pushed to the server, budget consumpion
+		and CBS thread switching in/out of the CPU. This option
+		requires CONFIG_CBS to be included within the project.
+
+if CBS_LOG
+menu "CBS events to be logged"
```

Comment:
The events being recorded here really seem like they would fit better with Zephyr's tracing subsystem. Can they use that mechanism instead?